scripts/batchProcessRasters.R

```diff
@@ -144,102 +149,273 @@ batchImportRasters_toOSW<-function(inpJSON){
+					errmsg<-toJSON(svb)
+					retJSON<-paste0("{\"Out\":[{\"Result\":\"Error\",\"Process\":\"Save object in bucket\",\"value\":\"Failed to save data table in the bucket\"}], \"Description\":\"",errmsg,"\"}")
+					if(flog==0){
+						cat("Processed raster but failed to save data table in the bucket.",sep ="\n",file = zz, append=TRUE)
+						if(logSessionInfo){
+							cat(sitd,sep="\n",file=zz,append=TRUE)
+						}
+						cat("\n","\n",file = zz, append=TRUE)
+						cat("Terminating batch raster processing.",sep ="\n",file = zz, append=TRUE)
+						close(zz)
+						svlg<-try(put_object(file=logfile,object=savelog,bucket="offshore-wind-data"),silent=TRUE)
+					}
+					return(retJSON)
+				}
+				
+				## Tabulate date of creation
+				recId<-as.character(tablesToProcess[ii,"recId"])
+				season<-as.character(tablesToProcess[ii,"Season"])
+				dateVal<-format(Sys.time(),"%Y-%m-%d %H:%M")
+				if(season=="Summer"){  #Can only be one of four...
+					newdata<-list(SummerTableVer=dateVal)
+				}else if(season=="Fall"){
+					newdata<-list(FallTableVer=dateVal)
+				}else if(season=="Winter"){
+					newdata<-list(WinterTableVer=dateVal)
+				}else{
+					newdata<-list(SpringTableVer=dateVal)
+				}
+				
+				#Update the AT to append the date
+				appendtry<-try(air_update(base="appO3EgtQh3UREeun",table_name=atTable,
+								record_id=recId,record_data=newdata),silent=TRUE)
+				if(inherits(appendtry,"try-error")){
+					tdf<-data.frame(Base=datasetNm,Layer=layerNm,Season=season,
+							ReprojectionResult="Failed to update metadata on raster in AirTable",
+							ATupdateResult=paste(appendtry,collapse="; "))
+					if(flog==0){
+						cat("Failed to update metadata on raster in AirTable",sep ="\n",file = zz, append=TRUE)
+					}
+				}else{
+					tdf<-data.frame(Base=datasetNm,Layer=layerNm,Season=season,
+							ReprojectionResult="Successfully reprojected and saved",
+							ATupdateResult="Successfully updated AT record")
+					if(flog==0){
+						cat("Successfully reprojected and saved the raster.",sep ="\n",file = zz, append=TRUE)
+					}
+				}
+			}else{
+				tdf<-data.frame(Base=datasetNm,Layer=layerNm,Season=season, 
+						ReprojectionResult=paste(sppTable, collapse="; "),
+						ATupdateResult="Failed to re-project raster to aliquote grid")
+				if(flog==0){
+					cat("Failed to re-project raster to aliquote grid.",sep ="\n",file = zz, append=TRUE)
+				}
+			}
+		}else{ #failed to read the raster layer
+			tdf<-data.frame(Base=datasetNm,Layer=layerNm,Season=season,
+					ReprojectionResult="Failed to load the raster layer",
+					ATupdateResult=paste(rastLayer,collapse="; "))
+			if(flog==0){
+				cat("Failed to load the raster layer.",sep ="\n",file = zz, append=TRUE)
+			}
+		}
+		updateResults<-rbind(updateResults,tdf)
+	}
+	retJSON<-toJSON(updateResults)	
+	#retJSON<-paste0("{\"Out\":[{\"Result\":\"Success\",\"Process\":\"Everything but processing\",\"value\":\"Checks passed\"}], \"Description\":\"All good\"}")

```

Comment:
this will be in the commit history, just delete it.

Reply:
Done